evm_arithmetization/src/byte_packing/columns.rs

```diff
@@ -1,43 +1,48 @@
-pub(crate) const IS_READ: usize = 0;
-
-pub(super) const LEN_INDICES_START: usize = IS_READ + 1;
-// There are `NUM_BYTES` columns used to represent the length of
-// the input byte sequence for a (un)packing operation.
-// index_len(i) is 1 iff the length is i+1.
-pub(crate) const fn index_len(i: usize) -> usize {
-    debug_assert!(i < NUM_BYTES);
-    LEN_INDICES_START + i
+use std::mem::transmute;
+
+use zk_evm_proc_macro::Columns;
+
+use crate::{byte_packing::NUM_BYTES, util::indices_arr};
+
+/// A view of `BytePackingStark`'s columns.
+#[repr(C)]
+#[derive(Columns, Eq, PartialEq, Debug)]
+pub(crate) struct BytePackingColumnsView<T: Copy> {
+    /// 1 if this is a READ operation, and 0 if this is a WRITE operation.
+    pub is_read: T,
+
+    // There are `NUM_BYTES` columns used to represent the length of
+    // the input byte sequence for a (un)packing operation.
+    // index_len[i] is 1 iff the length is i+1.
+    pub index_len: [T; NUM_BYTES],
+
+    pub addr_context: T,
+    pub addr_segment: T,
+    pub addr_virtual: T,
+    pub timestamp: T,
+
+    // 32 byte limbs hold a total of 256 bits.
+    // There are `NUM_BYTES` columns used to store the values of the bytes
+    // that are being read/written for an (un)packing operation.
+    pub value_bytes: [T; NUM_BYTES],
+
+    /// The counter column (used for the logUp range check) starts from 0 and
+    /// increments.
+    pub range_counter: T,
+    /// The frequencies column used in logUp.
+    pub rc_frequencies: T,
 }
 
-// Note: Those are used to obtain the length of a sequence of bytes being
-// processed.
-pub(crate) const LEN_INDICES_COLS: Range<usize> = LEN_INDICES_START..LEN_INDICES_START + NUM_BYTES;
-
-pub(crate) const ADDR_CONTEXT: usize = LEN_INDICES_START + NUM_BYTES;
-pub(crate) const ADDR_SEGMENT: usize = ADDR_CONTEXT + 1;
-pub(crate) const ADDR_VIRTUAL: usize = ADDR_SEGMENT + 1;
-pub(crate) const TIMESTAMP: usize = ADDR_VIRTUAL + 1;
-
-// 32 byte limbs hold a total of 256 bits.
-const BYTES_VALUES_START: usize = TIMESTAMP + 1;
-// There are `NUM_BYTES` columns used to store the values of the bytes
-// that are being read/written for an (un)packing operation.
-pub(crate) const fn value_bytes(i: usize) -> usize {
-    debug_assert!(i < NUM_BYTES);
-    BYTES_VALUES_START + i
-}
+// `u8` is guaranteed to have a `size_of` of 1.
+/// Number of columns in `BytePackingStark`.
+pub(crate) const NUM_COLUMNS: usize = size_of::<BytePackingColumnsView<u8>>();
 
-/// The counter column (used for the range check) starts from 0 and increments.
-pub(crate) const RANGE_COUNTER: usize = BYTES_VALUES_START + NUM_BYTES;
-/// The frequencies column used in logUp.
-pub(crate) const RC_FREQUENCIES: usize = RANGE_COUNTER + 1;
+const fn make_col_map() -> BytePackingColumnsView<usize> {
+    let indices_arr = indices_arr::<NUM_COLUMNS>();
+    unsafe { transmute::<[usize; NUM_COLUMNS], BytePackingColumnsView<usize>>(indices_arr) }
```

Comment:
IIUC this creates a struct from a list of offsets.  I see it's done the same way in `memory/columns.rs`, but wondering if this is the best way to construct a struct statically?

Reply:
My second thought is that whether it should be done this way or an alternative way is actually out-of-scope for this PR.  It should really just implement it in the same way as the other starks, and then we can make a separate issue on using a alternative approach this if desired. Thus, in conclusion: lgtm.